fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/python/PythonUtils.xtend

```diff
@@ -626,6 +627,42 @@ class PythonUtils {
             return "";
     }
     
+    def String pythonAttributeVariableNameHL(Attribute attr) {
+    	var attrVariableName = attr.name
+    	var indexList = new BasicEList<Integer>
+    	var ch =0 
+    	var stringSize = (attr.name.length -1)
+    	for (ch =0; ch < attrVariableName.length; ch++){
+    		if(Character.isUpperCase(attrVariableName.charAt(ch))){
+    			indexList.add(ch);
+    		}
+    	}
+    	var checkNxtIndex = 0
+    	var position = 0
+    	var count = 0
+    	for (index:indexList){
+    		if(checkNxtIndex !== stringSize)
+    		{   			
+	    		if(index==checkNxtIndex){
+	    			checkNxtIndex = (index +1)
+	    			if(Character.isLowerCase(attr.name.charAt((checkNxtIndex)))){
+	    				position = (index + count)
+	    				attrVariableName = attrVariableName.substring(0, position)+ '_' + attrVariableName.substring(position)
+	    				count+=1
+	    			}
+	    		}
+	    		else{
+	    			position = (index + count)
+	    			attrVariableName = attrVariableName.substring(0, position)+ '_' + attrVariableName.substring(position)
+	    			checkNxtIndex = (index +1)
+	    			count+=1
+	    		}
+    		}
+    	} 
+		return attrVariableName.toLowerCase
+	}
```

Comment:
The indentation of lines 661 to 663 makes it hard to read.  E.g., the return is outside of the for loop, but it is indented.

```java
    def String pythonAttributeVariableNameHL(Attribute attr) {
    	var attrVariableName = attr.name
    	var indexList = new BasicEList<Integer>
    	var ch =0 
    	var stringSize = (attr.name.length -1)
    	for (ch =0; ch < attrVariableName.length; ch++){
    		if(Character.isUpperCase(attrVariableName.charAt(ch))){
    			indexList.add(ch);
    		}
    	}
    	var checkNxtIndex = 0
    	var position = 0
    	var count = 0
    	for (index:indexList){
    		...
    	} 
    	return attrVariableName.toLowerCase
    }
```

Reply:
+1

Reply:
Okay Thanks. I will update.

Reply:
Updated with commit https://github.com/tango-controls/pogo/pull/87/commits/578e2966f9cc014086334e4e5385f91e58ffb94e